src/__tests__/core/block.transactionFilter.test.ts

```diff
@@ -89,6 +89,78 @@ describe('transactionFilter on Block->transactions', async () => {
     expect(result.errors).toHaveLength(1);
     expect(result.errors[0].message).toMatch(/^Expected type TransactionFilter/);
   });
+
+  test('contractCreation: true value, returns only transactions with contract creations', async () => {
+    const query = `
+      {
+        block(number: 4180981) {
+          transactions(filter: { contractCreation: true }) {
+            hash
+          }
+        }
+      }
+    `;
+
+    const expected = { data: { block: { transactions: [
+      {
+        'hash': '0xab8565481d5edcd162da315bf194b349451b37efbe10e068c601a8cfa84728ca'
+      }
+    ] } } };
+    const result = await execQuery(query);
+    expect(result).toEqual(expected);
+  });
+
+  test('contractCreation: false value, returns only transactions without contract creations', async () => {
+    const query = `
+      {
+        block(number: 5818471) {
+          transactions(filter: { contractCreation: false }) {
+            hash
```

Comment:
I would expect you to select the `to->address` field and assert that it is filled in all cases.

Reply:
SGTM